step-4/client/app/components/withredux/withredux.component.js

```diff
@@ -66,9 +66,13 @@ export const WithReduxComponent = {
             doneTodosByType: TodosSelectors.getDoneTodosByTypeGenerator(typeId)(state),
             errorTodosByType: TodosSelectors.getErrorTodosByTypeGenerator(typeId)(state),
 
+            // Parametric Selectors to list and delete
```

Comment:
Maybe we can void polluting the state to this map with the $cache mechanism. Maybe providing our own mapping function that would hide the cache issues under the hood.